### PR TITLE
fix: Safari space distribution glitch

### PR DESCRIPTION
**Pull request by marks0351**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=0_Nnh2L)
## Description

we use dynamic transition values on the zone widget to provide smooth effect when a zone is resized via space distribution.
we recently made changes of how we calculate(`convertFlexGrowToFlexBasis`) flex-basis values becuase we had introduced gap between zones via tokens. for this we had used calc.

dynamic transition and calc css values don't seem work smooth in conjunction in safari. making it a glitch experience.
hence disabling transition in safari until I figure how to get around this.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8936966895>
> Commit: db551717a9cb12fc3aaa029574c70b650b6fdc7c
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8936966895&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export interface SpaceDistributionZoneDomCollection {
   leftZonePropPaneDom: HTMLElement | null;
   rightZonePropPaneDom: HTMLElement | null;
 }
-
+const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
```
[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts]: have made changes to use an inhouse function now.

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts [hunk 2]
```diff
@@ -96,9 +102,9 @@ const applyResistiveForceOnHandleMove = (
   // Check if the zones hit the minimum limit
   const hasHitMinimumLimit =
     isLeftZoneLessThanMinimum || isRightZoneLessThanMinimum;
-
+  const enableTransition = !isSafari() && hasHitMinimumLimit;
```
Comment (KelvinOm) [inline app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts:105]: (nitpick) 
```suggestion
  const isTransitionEnabled = !isSafari() && hasHitMinimumLimit;
```

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts [hunk 3]
```diff
@@ -37,11 +42,12 @@ const adjustZoneFlexGrowForMagneticEffect = (
   const reflectOnPropPane = leftZonePropPaneDom && rightZonePropPaneDom;
 
   // Calculate transition duration based on mouse speed
-  const transitionStyle = `all ${
-    baseAnimationDuration -
-    Math.min(mouseSpeed, speedLimitForAnimation) * ratioOfSpeedToAnimation
-  }s ease-in-out`;
-
+  const transitionStyle = isSafari()
```
Comment (KelvinOm) [inline app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts:45]: Which tells me that @tbrizitsky won't be happy with this changes. Check this with him please.

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts [hunk 4]
```diff
@@ -37,11 +42,12 @@ const adjustZoneFlexGrowForMagneticEffect = (
...
+    ? ""
+    : `all ${
```
Comment (KelvinOm) [inline app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts:47]: Do we really need `all` here? Can we specify only those properties that we will change?

File: app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts (hunk 3 above)
Comment (marks0351) [reply app/client/src/layoutSystems/anvil/sectionSpaceDistributor/utils/onMouseMoveUtils.ts:45]: yep had discussed this with him.

**Pull request merged**